### PR TITLE
Issue 224: Ensure Claim does not fail even when group ID changes 

### PR DESCRIPTION
**Pull request by anandrgitnirman**
https://github.com/singnet/snet-daemon/issues/224
Ensure Group ID check should only apply at the time of storing channel state in etcd for the first time
Moved the group ID check from BlockchainChannelReader to lockingPaymentChannelService.PaymentChannel()

### CODE REVIEW

**Comment by anandrgitnirman** [comment]
@astroseger , tested this , works fine with claim now 

@vsbogd , request your review on this

File: escrow/escrow.go [hunk 1]
```diff
@@ -56,6 +66,15 @@ func (h *lockingPaymentChannelService) PaymentChannel(key *PaymentChannelKey) (c
 	return MergeStorageAndBlockchainChannelState(storageChannel, blockchainChannel), true, nil
 }
 
+//Check if the channel belongs to the same group Id
+func (h *lockingPaymentChannelService) VerifyGroupId(configGroupID [32]byte ,blockChainGroupID  [32]byte ) error {
```
Comment (vsbogd) [inline escrow/escrow.go]: I would rename the method to `verifyGroupId` to make it private.

File: escrow/escrow_test.go [hunk 2]
```diff
@@ -280,3 +281,41 @@ func (suite *PaymentChannelServiceSuite) TestStartClaim() {
 	assert.Equal(suite.T(), suite.payment(), claim.Payment())
 	assert.Equal(suite.T(), []*Payment{suite.payment()}, claims)
 }
+
+func (suite *PaymentChannelServiceSuite) TestVerifyGroupId() {
+
+
+	service := NewPaymentChannelService(
```
Comment (vsbogd) [inline escrow/escrow_test.go]: Test could be simplified and made more readable by copying `suite.service` and replacing only field required to replace:
```golang
service := suite.service()
service.(lockingPaymentChannelService).blockchainReader.replicaGroupID = func(channelID *big.Int) (*blockchain.MultiPartyEscrowChannel, bool, error) {
    return suite.mpeChannel(), true, nil
}
```

File: escrow/escrow_test.go [hunk 3]
```diff
@@ -280,3 +281,41 @@ func (suite *PaymentChannelServiceSuite) TestStartClaim() {
 	assert.Equal(suite.T(), suite.payment(), claim.Payment())
 	assert.Equal(suite.T(), []*Payment{suite.payment()}, claims)
 }
+
+func (suite *PaymentChannelServiceSuite) TestVerifyGroupId() {
```
Comment (vsbogd) [inline escrow/escrow_test.go]: :+1:

File: escrow/escrow.go [hunk 4]
```diff
@@ -45,8 +45,18 @@ func (h *lockingPaymentChannelService) PaymentChannel(key *PaymentChannelKey) (c
...
 	}
 
+
 	blockchainChannel, blockchainOk, err := h.blockchainReader.GetChannelStateFromBlockchain(key)
+
 	if !storageOk {
+		//Group ID check is only done for the first time , when the channel is added to storage from the block chain ,
+		//if the channel is already present in the storage the group ID check is skipped.
+	    if blockchainChannel != nil {
+			blockChainGroupID,err := h.blockchainReader.replicaGroupID()
```
Comment (vsbogd) [inline escrow/escrow.go]: As `replicaGroupId` is not used in blockchain reader anymore I would move it into `lockingPaymentChannelService` itself. But it will probably require fixing unit tests, so it makes sense to do separate PR for this.

File: escrow/escrow_test.go (hunk 2 above)
Comment (anandrgitnirman) [reply escrow/escrow_test.go]: Thansk @vsbogd  , fixed this

File: escrow/escrow.go (hunk 4 above)
Comment (anandrgitnirman) [reply escrow/escrow.go]: Will submit a separate PR by tomorrow , hopefully by then this is approved :)
